src/rgw/rgw_sal_motr.cc

```diff
@@ -3853,16 +3853,16 @@ std::string MotrStore::get_cluster_id(const DoutPrefixProvider* dpp,  optional_y
 }
 
 int MotrStore::init_metadata_cache(const DoutPrefixProvider *dpp,
-                                   CephContext *cct)
+                                   CephContext *cct, bool use_cache)
 {
   this->obj_meta_cache = new MotrMetaCache(dpp, cct);
```

Comment:
In my opinion we should create instances of the cache objects (for object metadata, user & bucket instances only when use_cache is true.

Reply:
Yes, I wanted to do the same earlier, but look at the way we are calling methods on these objects.
```c++
  if (store->get_bucket_inst_cache()->get(dpp, info.bucket.name, bl)) {
    // Cache misses.
    ldpp_dout(dpp, 20) << "load_bucket(): name=" << info.bucket.name << dendl;
    int rc = store->do_idx_op_by_name(RGW_MOTR_BUCKET_INST_IDX_NAME,
                                      M0_IC_GET, info.bucket.name, bl);
    ldpp_dout(dpp, 20) << "load_bucket(): rc=" << rc << dendl;
    if (rc < 0)
      return rc;
    store->get_bucket_inst_cache()->put(dpp, info.bucket.name, bl);
  }
```
here if keep the `bucket_cache_obj = nullptr` let's say then calling `->get()` on the same will result in an exception. 
We can add checks before calling get or put. but this is in all the methods so it will result in a lot of changes.

Reply:
Only 23 instances though. we can fix this now if we want to. cause in future it will grow only..
![image](https://user-images.githubusercontent.com/45289079/158111443-10a48c8d-75ab-4205-96dc-ddaf53b28b9e.png)

Reply:
Lets not make lot of changes. So current changes are fine